x/xcvm/keeper/transfer_intent_test.go

```diff
@@ -0,0 +1,619 @@
+package keeper_test
+
+import (
+	"crypto/ecdsa"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/types/address"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	wasmtypes "github.com/cosmos/ibc-go/v7/modules/light-clients/08-wasm/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/trie"
+	prysmtypes "github.com/prysmaticlabs/prysm/v4/proto/eth/v1"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	tmproto "github.com/cometbft/cometbft/proto/tendermint/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/notional-labs/composable/v6/app"
+	"github.com/notional-labs/composable/v6/app/helpers"
+	"github.com/notional-labs/composable/v6/x/xcvm/types"
+	"github.com/stretchr/testify/suite"
+)
+
+type TransferIntentTestSuite struct {
+	suite.Suite
+
+	ctx        sdk.Context
+	app        *app.ComposableApp
+	wasmCodeId []byte
+	moduleAddr sdk.AccAddress
+	govAddr    sdk.AccAddress
+}
+
+func (suite *TransferIntentTestSuite) SetupTest() {
+	suite.app = helpers.SetupComposableAppWithValSet(suite.T())
+	suite.ctx = suite.app.BaseApp.NewContext(false, tmproto.Header{Height: 1, ChainID: "centauri-1", Time: time.Now().UTC()})
+
+	// get wasm ethereum light client
+	wasmBytes, err := os.ReadFile("testdata/icsxx_ethereum_cw.wasm")
```

Comment:
add a comment that references the `icsxx_ethereum_cw` commit hash used for this test